src/ui/image.cpp

```diff
@@ -211,9 +214,46 @@ ImageBuffer *get_image(unsigned bid) {
   return result;
 }
 
+void scaleImage(ImageBuffer* image, var_num_t scaling) {
+  if(scaling == 1.0 || scaling <= 0.0) {
+    return;
+  }
+
+  uint16_t xx, yy, w, h;
+  uint32_t offsetImage, offsetScaledImage;
+
+  w = round((var_num_t)image->_width * scaling);
+  h = round((var_num_t)image->_height * scaling);
+
+  uint8_t* scaledImage = (uint8_t *)malloc(w * h * 4);
+  if(!scaledImage) {
+    err_throw(ERR_IMAGE_LOAD, "Failed to allocate RAM");
```

Comment:
This sets up for SmallBASIC try/catch handling but isn't itself a `throw` like with java etc.
So you still need to avoid invoking the code below. So either use `else { ...` or `return` here.
In this case a simple return is the simplest solution.

Reply:
Thanks, I didn't know that. I changed it.